packages/react-editor/src/use-text-length.tsx

```diff
@@ -6,7 +8,15 @@ export function useTextLength({
   maxByteLength: number;
 }) {
   const text = getText();
-  const lengthInBytes = new TextEncoder().encode(text).length;
+
+  // Mentions don't occupy space in the cast, so we need to ignore them for our length calculation
+  const structuredTextUnits = convertCastPlainTextToStructured({ text });
```

Comment:
lgtm! this fn could be a little slow to run on each keydown as the regexes are pretty involved - did you feel any input delay?

Reply:
nah i don't feel any input delay, don't think it will be a problem